### PR TITLE
fix(top-app-bar): reopen short top-app-bar

### LINKED ISSUE #4928: TopAppBar short never reopens after initial scroll
### Steps to reproduce
1. Go to https://material.io/develop/web/components/top-app-bar/
2. Scroll down to the "short" variant of the top-app-bar
3. Scroll down, notice it shrinks
4. Scroll back to top, notice it does not restore to its original full size

### Expected behavior
The top-app-bar should restore to its original size for the short variant whenever the page is scrolled back to the top

Noticed this while upgrading RMWC to 3.0.0. I don't believe this was an issue with 2.0.0

### PR DESCRIPTION
**Pull request by TroyTae**
I don't know why, but `Semantic Pull Request` is pending continuously.
So I reopen this PR (Original PR -> https://github.com/material-components/material-components-web/pull/4959)

fixes #4928

### CODE REVIEW

**Comment by TroyTae** [comment]
@abhiomkar 
Please give us feedback if you need to work on it additionally.

File: test/unit/mdc-top-app-bar/short.foundation.test.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ test('short top app bar: scrollHandler does not call getViewportScrollY method '
   const {foundation, mockAdapter} = setupTest();
   td.when(mockAdapter.hasClass(MDCTopAppBarFoundation.cssClasses.SHORT_COLLAPSED_CLASS)).thenReturn(true);
   foundation.init();
-  td.verify(mockAdapter.getViewportScrollY(), {times: 0});
+  td.verify(mockAdapter.getViewportScrollY(), {times: 1});
```
Comment (abhiomkar) [inline test/unit/mdc-top-app-bar/short.foundation.test.js]: Update test description accordingly.

**Comment by TroyTae** [comment]
move to #4995

File: test/unit/mdc-top-app-bar/short.foundation.test.js (hunk 1 above)
Comment (TroyTae) [reply test/unit/mdc-top-app-bar/short.foundation.test.js]: Done!
And I have one question...
If I want to request PR confirmation, I need to annotate some maintainer?
I'm used to MDCReact so I just leave PR until 1 week.

**Pull request closed without merging**